cohortextractor/tpp_backend.py

```diff
@@ -2787,6 +2788,207 @@ def patients_outpatient_appointment_date(
               OPA.Patient_ID
             """
 
+    def create_therapeutics_table(self):
+        """
+        Create a temporarary Therapeutics table to use for `with_covid_therapeutics` queries
+        All columns in the Therapeutics table are VARCHAR.  This casts the two date columns that
+        we use in queries to DATE type, and adds case insensitive collation to the Interventaion
+        and CurrentStatus columns.
+        All columns are included, including those that we don't use, so that when we remove
+        duplicates, we only remove complete duplicate rows
+        """
+        if self._therapeutics_table_name is None:
```

Comment:
Building a temp table for the just this data to get it into the format we want is a neat approach